contracts/protocols/mainnet/compound_iii/helpers.sol

```diff
@@ -0,0 +1,387 @@
+        uint64 indexScale;
+        ///@dev An index for tracking participation of accounts that supply the base asset.
+        uint64 trackingSupplyIndex;
+        ///@dev An index for tracking participation of accounts that borrow the base asset.
+        uint64 trackingBorrowIndex;
+    }
+
+    struct Scales {
+        ///@dev liquidation factor, borrow factor scale
+        uint64 factorScale;
+        ///@dev scale for USD prices
+        uint64 priceScale;
+        ///@dev The scale for the index tracking protocol rewards, useful in calculating rewards APR
+        uint64 trackingIndexScale;
+    }
+
+    struct Token {
+        uint8 offset;
+        address token;
+        string symbol;
+        ///@dev 10**decimals
+        uint256 scale;
+    }
+
+    struct AssetData {
+        Token token;
+        ///@dev token's priceFeed
+        address priceFeed;
+        ///@dev answer as per latestRoundData from the priceFeed scaled by priceScale
+        uint256 price;
+        ///@dev 10**decimals
+        uint64 scale;
+        ///@dev The collateral factor(decides how much each collateral can increase borrowing capacity of user),
+        //integer representing the decimal value scaled up by 10 ^ 18.
+        uint64 borrowCollateralFactor;
+        ///@dev sets the limits for account's borrow balance,
+        //integer representing the decimal value scaled up by 10 ^ 18.
+        uint64 liquidateCollateralFactor;
+        ///@dev liquidation penalty deducted from account's balance upon absorption
+        uint64 liquidationFactor;
+        ///@dev integer scaled up by 10 ^ decimals
+        uint128 supplyCap;
+        ///@dev  current amount of collateral that all accounts have supplied
+        uint128 totalCollateral;
+    }
+
+    struct AccountFlags {
+        bool isLiquidatable;
+        bool isBorrowCollateralized;
+    }
+
+    struct MarketFlags {
+        bool isAbsorbPaused;
+        bool isBuyPaused;
+        bool isSupplyPaused;
+        bool isTransferPaused;
+        bool isWithdrawPaused;
+    }
+
+    struct UserCollateralData {
+        Token token;
+        uint256 suppliedBalance;
+    }
+
+    struct RewardsConfig {
+        address token;
+        uint64 rescaleFactor;
+        bool shouldUpScale;
+        uint104 baseMinForRewards;
+    }
+
+    struct UserRewardsData {
+        address rewardToken;
+        uint256 amountOwed;
+        uint256 amountClaimed;
+    }
+
+    struct UserData {
+        int104 baseBalance;
+        //the base balance of supplies with interest, 0 for borrowing case or no supplies
+        uint256 suppliedBalance;
+        //the borrow base balance including interest, for non-negative base asset balance value is 0
+        uint256 borrowedBalance;
+        uint16 assetsIn;
+        uint64 accountTrackingIndex;
+        uint64 interestAccrued;
+        uint256 userNonce;
+        int256 borrowableAmount;
+        UserRewardsData rewards;
```

Comment:
I think there can be multiple rewards possible for a market, so it should be array.